drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -143,7 +143,11 @@ static int rtc_stm32_set_alarm(struct device *dev, u8_t chan_id,
 	data->user_data = alarm_cfg->user_data;
 
 	if ((alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE) == 0) {
-		ticks += now;
+		/* Add +1 in order to compensate the partially started tick. */
+		/* Alarm will expire between requested ticks and ticks+1. */
+		/* In case only 1 tick is requested, it will avoid */
+		/* that tick+1 event occurs before alarm setting is finished. */
```

Comment:
Use a continuous comment,

```c
/* a long
 * comment
 */
```
I guess that by doing this adjustment alarm will always be triggered in `ticks` plus the elapsed amount of the potentially started tick. So in the worst case, user will experience 1 tick more and in the best case the exact amount of ticks. If this adjustment is not done, user will experience in the worst case 1 tick less and again the exact amount of ticks for best case. Assuming setting takes < 1s, the 1s case could be treated differently so other cases are not affected, right?

Reply:
> Use a continuous comment

Done

Yes it is possible to handle the special case of 1tick request and let other case unmodified.
But if I add +1 tick only when 1 tick is requested, it means that requesting 1Tick will be the same than requesting 2 ticks.
Also I think normal (more intuitive) that when application request N ticks we guaranntee at least N ticks and not less.

Reply:
Ok, fair point.